server/core/src/actors/v1_write.rs

```diff
@@ -1108,6 +1108,7 @@ impl QueryServerWriteV1 {
             "class".into(),
             Value::new_class("posixgroup"),
         )))
+        .chain(iter::once(Some(Modify::Purged("gidnumber".into()))))
```

Comment:
What happens if the user changes another attribute but not the gid here now? I think this purge needs to be associated only if a gid is actually set too.

Reply:
I think I should move it inside the map then

Reply:
Or just restructure the code to make it clearer what the intent is.